crates/evm/src/system_calls/eip2935.rs

```diff
@@ -23,19 +24,13 @@ use revm_primitives::ResultAndState;
 ///
 /// [EIP-2935]: https://eips.ethereum.org/EIPS/eip-2935
 #[inline]
-pub(crate) fn transact_blockhashes_contract_call<EvmConfig, EXT, DB>(
-    evm_config: &EvmConfig,
+pub(crate) fn transact_blockhashes_contract_call(
     chain_spec: impl EthereumHardforks,
     block_timestamp: u64,
     block_number: u64,
     parent_block_hash: B256,
-    evm: &mut Evm<'_, EXT, DB>,
-) -> Result<Option<ResultAndState>, BlockExecutionError>
-where
-    DB: Database,
-    DB::Error: core::fmt::Display,
-    EvmConfig: ConfigureEvm,
-{
+    evm: &mut impl Evm<Error: Display>,
```

Comment:
we could now also introduce an EthEvmExt trait that exposes these directly on `Evm`